app-modules/assistant/src/Filament/Pages/PersonalAssistant.php

```diff
@@ -273,6 +277,77 @@ public function editChatAction(): Action
             ]);
     }
 
+    public function shareChatAction(): Action
+    {
+        return Action::make('shareChat')
+            ->modalSubmitActionLabel('Share')
+            ->modalWidth('md')
+            ->size(ActionSize::ExtraSmall)
+            ->form([
+                Select::make('target_type')
+                    ->label('Type')
+                    ->options([
+                        'team' => 'Team',
+                        'user' => 'User',
+                    ])
+                    ->default('user')
+                    ->required()
+                    ->selectablePlaceholder(false)
+                    ->live(),
+                Select::make('target_ids')
+                    ->label('Targets')
+                    ->options(fn (Get $get): Collection => match ($get('target_type')) {
+                        'team' => Team::orderBy('name')->pluck('name', 'id'),
+                        'user' => User::whereKeyNot(auth()->id())->orderBy('name')->pluck('name', 'id'),
+                        default => collect(),
+                    })
+                    ->searchable()
+                    ->multiple()
+                    ->required(),
+            ])
+            ->action(function (array $arguments, array $data) {
+                $chat = AssistantChat::find($arguments['chat']);
+
+                $users = match ($data['target_type']) {
+                    'team' => collect($data['target_ids'])
+                        ->map(fn ($id) => Team::find($id)->users()->whereKeyNot(auth()->id())->get())
+                        ->flatten()
+                        ->unique(),
+                    'user' => User::whereIn('id', $data['target_ids'])->get(),
+                };
+
+                $users
+                    ->each(
+                        function (User $user) use ($chat) {
+                            $replica = $chat
+                                ->replicate(['id', 'user_id'])
+                                ->user()->associate($user);
+
+                            $replica->save();
+
+                            //TODO: notification?
```

Comment:
Yup, let's add a notification here to indicate success to the user after replication.